app/src/main/java/be/huffle/allowanceplanner/activities/AddExpenseActivity.java

```diff
@@ -30,51 +31,48 @@ protected void onCreate(Bundle savedInstanceState)
 		setContentView(R.layout.activity_add);
 
 		ActionBar actionBar = getSupportActionBar();
-		Button addButton = findViewById(R.id.add_button);
 		amountEditText = findViewById(R.id.amount_input);
 		descriptionEditText = findViewById(R.id.description_input);
 		sharedPreferences = getApplicationContext().getSharedPreferences("SharedPref", MODE_PRIVATE);
 		editor = sharedPreferences.edit();
 
 		actionBar.setDisplayHomeAsUpEnabled(true);
 		actionBar.setTitle("Add Expense");
-
-		addButton.setOnClickListener(this);
 	}
 
 	@Override
 	public boolean onOptionsItemSelected(@NonNull MenuItem item)
 	{
-		int id = item.getItemId();
-
-		if (id == android.R.id.home)
+		switch (item.getItemId())
 		{
-			finish();
-		}
-
-		return true;
-	}
-
-	@Override
-	public void onClick(View view)
-	{
-		switch (view.getId())
-		{
-			case R.id.add_button:
+			case android.R.id.home:
```

Comment:
```suggestion
			case R.id.home:
```

Reply:
_android_ needs to stay for the back button to work